core/client/src/main/java/alluxio/client/file/FileSystemContext.java

```diff
@@ -42,27 +58,146 @@
  * so, because thread A holds the lock on {@link FileSystemContext}.
  */
 @ThreadSafe
-public enum FileSystemContext {
-  INSTANCE;
+public class FileSystemContext {
+  public static final FileSystemContext INSTANCE = new FileSystemContext(null);
```

Comment:
i can send another PR to rename it. If you don't feel very strong about this, I will just use INSTANCE for now. Too many places are affected.